benches/src/access_control.rs

```diff
@@ -65,51 +65,17 @@ pub async fn bench() -> eyre::Result<()> {
         ("setRoleAdmin(ROLE, NEW_ADMIN_ROLE)", receipt!(contract.setRoleAdmin(ROLE.into(), NEW_ADMIN_ROLE.into()))?),
     ];
 
-    // Calculate the width of the longest function name.
-    let max_name_width = receipts
-        .iter()
-        .max_by_key(|x| x.0.len())
-        .expect("should at least bench one function")
-        .0
-        .len();
-    let name_width = max_name_width.max("AccessControl".len());
-
-    // Calculate the total width of the table.
-    let total_width = name_width + 3 + 6 + 3 + 6 + 3 + 20 + 4; // 3 for padding, 4 for outer borders
-
-    // Print the table header.
-    println!("+{}+", "-".repeat(total_width - 2));
-    println!(
-        "| {:<width$} | L2 Gas | L1 Gas |        Effective Gas |",
-        "AccessControl",
-        width = name_width
-    );
-    println!(
-        "|{}+--------+--------+----------------------|",
-        "-".repeat(name_width + 2)
-    );
-
-    // Print each row.
-    for (func_name, receipt) in receipts {
+    let mut report = Report::new("AccessControl");
+    for (signature, receipt) in receipts {
```

Comment:
IMHO this loop is repeated in each benchmark; Can we have a common code for it? If not, it is fine to leave as it is.

Reply:
yes, that's a second refactor!

Reply:
Maybe implementing `From<T>` trait for `Report`? I mean `Report::from(receipts)`

Reply:
We can't cause we need the contract name.

Reply:
Missed that one.